android-tests/device-tests/build.gradle.kts

```diff
@@ -9,7 +9,7 @@ android {
     compileSdk = 35
```

Comment:
we don't need to change the compileSdk: this way we are allowed to compile using a recent Java version; `minSdk` will ensure it works on the lower target (in our case 28). In other words, this is similar to java.version vs java.release